boot/bootutil/src/bootutil_priv.h

```diff
@@ -140,7 +140,26 @@ struct boot_status {
  *      (`MCUBOOT_ENC_IMAGES`).
  */
 
-extern const uint32_t boot_img_magic[4];
+union boot_img_magic_t
+{
+    struct {
```

Comment:
It wouldn't hurt to add the packed attribute here, but it is not needed in this case since the alignment requirement for the inner struct is **2** imposed by the `uint16_t` member. So the compiler won't add padding to this region.
Besides, the static assertion on line **162** will ensure we won't get caught in any misbehavior of the compiler.